go/enclave/enclave.go

```diff
@@ -700,84 +704,83 @@ func (e *enclaveImpl) Stop() error {
 	}
 
 	callMsg, err := gethencoding.ExtractEthCall(paramList[0])
 	if err != nil {
-		return nil, fmt.Errorf("unable to decode EthCall Params - %w", err)
+		err = fmt.Errorf("unable to decode EthCall Params - %w", err)
+		return responses.AsPlaintextError(err)
 	}
 
 	// encryption will fail if From address is not provided
 	if callMsg.From == nil {
-		return nil, fmt.Errorf("no from address provided")
+		err = fmt.Errorf("no from address provided")
+		return responses.AsPlaintextError(err)
 	}
 
+	encryptor := e.rpcEncryptionManager.CreateEncryptorFor(*callMsg.From)
+
 	// extract optional block number - defaults to the latest block if not avail
 	blockNumber, err := gethencoding.ExtractOptionalBlockNumber(paramList, 1)
 	if err != nil {
-		return nil, fmt.Errorf("unable to extract requested block number - %w", err)
+		err = fmt.Errorf("unable to extract requested block number - %w", err)
+		return responses.AsEncryptedError(err, encryptor)
 	}
 
 	// TODO hook the correct blockNumber from the API call (paramList[1])
 	gasEstimate, err := e.DoEstimateGas(callMsg, blockNumber, e.chain.GlobalGasCap)
 	if err != nil {
-		return nil, fmt.Errorf("unable to estimate transaction - %w", err)
+		err = fmt.Errorf("unable to estimate transaction - %w", err)
+		return responses.AsEncryptedError(err, encryptor)
 	}
 
-	// encrypt the gas cost with the callMsg.From viewing key
-	encryptedGasCost, err := e.rpcEncryptionManager.EncryptWithViewingKey(*callMsg.From, []byte(hexutil.EncodeUint64(uint64(gasEstimate))))
-	if err != nil {
-		return nil, fmt.Errorf("enclave could not respond securely to eth_estimateGas request. Cause: %w", err)
-	}
-	return encryptedGasCost, nil
+	return responses.AsEncryptedResponse(&gasEstimate, encryptor)
 }
 
-func (e *enclaveImpl) GetLogs(encryptedParams common.EncryptedParamsGetLogs) (common.EncryptedResponseGetLogs, error) {
+func (e *enclaveImpl) GetLogs(encryptedParams common.EncryptedParamsGetLogs) responses.Logs {
 	// We decrypt the params.
 	paramBytes, err := e.rpcEncryptionManager.DecryptBytes(encryptedParams)
 	if err != nil {
-		return nil, fmt.Errorf("unable to decrypt params in GetLogs request. Cause: %w", err)
+		err = fmt.Errorf("unable to decrypt params in GetLogs request. Cause: %w", err)
+		return responses.AsPlaintextError(err)
 	}
 
 	// We extract the arguments from the param bytes.
 	filter, forAddress, err := extractGetLogsParams(paramBytes)
 	if err != nil {
-		return nil, err
+		return responses.AsPlaintextError(err)
 	}
 
+	encryptor := e.rpcEncryptionManager.CreateEncryptorFor(*forAddress)
+
 	// We retrieve the relevant logs that match the filter.
 	filteredLogs, err := e.subscriptionManager.GetFilteredLogs(forAddress, filter)
 	if err != nil {
-		return nil, fmt.Errorf("could not retrieve logs matching the filter. Cause: %w", err)
+		err = fmt.Errorf("could not retrieve logs matching the filter. Cause: %w", err)
+		return responses.AsEncryptedError(err, encryptor)
```

Comment:
this one looks like a system error to me. Should be returned to the host as well.
I would assume an error is returned when something bad happened. Not when there are no logs.